next/test/visibility.test.ts

```diff
@@ -1,271 +1,267 @@
-        expect(businessNameField?.isVisible).toBe(false)
-
-        const personalValue: ObjectValue = { user: { type: 'personal' } as ObjectValue }
-        form.handleValidation(personalValue)
-        expect(personalNameField?.isVisible).toBe(true)
-        expect(businessNameField?.isVisible).toBe(false)
-
-        const businessValue: ObjectValue = { user: { type: 'business' } as ObjectValue }
-        form.handleValidation(businessValue)
-        expect(personalNameField?.isVisible).toBe(false)
-        expect(businessNameField?.isVisible).toBe(true)
-      }
-    })
-
-    it('should consider type errors in conditional fields', () => {
-      const schema: JsfObjectSchema = {
-        type: 'object',
-        properties: {
-          count: { type: 'number' },
-          details: { type: 'string' },
         },
         allOf: [
           {
             if: {
               properties: {
-                count: { minimum: 1 },
+                name: {
+                  const: userName,
+                },
               },
-              required: ['count'],
-            },
-            then: {
-              required: ['details'],
+              required: ['name'],
             },
           },
         ],
       }
 
-      const form = createHeadlessForm(schema)
-
-      const detailsField = form.fields.find(field => field.name === 'details')
-      expect(detailsField?.isVisible).toBe(false)
-
-      const invalidValue: ObjectValue = { count: 'not-a-number' }
-      form.handleValidation(invalidValue)
-      expect(detailsField?.isVisible).toBe(false)
-
-      const validValue: ObjectValue = { count: 2 }
-      form.handleValidation(validValue)
-      expect(detailsField?.isVisible).toBe(true)
+      it('should show the password field by default', () => {
+        const form = createHeadlessForm(schema)
+        // No name provided
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(true)
+
+        // Different name provided
+        form.handleValidation({
+          name: 'some name',
+        })
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(true)
+      })
+
+      it('should show the password field if the name is "admin"', () => {
+        const form = createHeadlessForm(schema)
+        form.handleValidation({
+          name: userName,
+        })
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(true)
+      })
     })
+  })
 
-    it('should handle multiple conditional rules affecting the same field', () => {
-      const schema: JsfObjectSchema = {
-        type: 'object',
+  // This does not work with v0 but I think it should work with v1
+  describe('if on object schema level', () => {
+    const schema: JsfObjectSchema = {
+      type: 'object',
+      properties: {
+        name: {
+          type: 'string',
+        },
+        password: {
+          type: 'string',
+        },
+      },
+      if: {
```

Comment:
suggestion: Can you move this to a `allOf` as that's the case for 99% of our internal forms :)

Reply:
yes 👍